Runtime/Tiles/RandomTerrainTile/RandomTerrainTile.cs

```diff
@@ -0,0 +1,349 @@
+            tileData.transform = Matrix4x4.identity;
+            tileData.color = Color.white;
+
+            int mask = TileValue(tileMap, location + new Vector3Int(0, 1, 0)) ? 1 : 0;
+            mask += TileValue(tileMap, location + new Vector3Int(1, 1, 0)) ? 2 : 0;
+            mask += TileValue(tileMap, location + new Vector3Int(1, 0, 0)) ? 4 : 0;
+            mask += TileValue(tileMap, location + new Vector3Int(1, -1, 0)) ? 8 : 0;
+            mask += TileValue(tileMap, location + new Vector3Int(0, -1, 0)) ? 16 : 0;
+            mask += TileValue(tileMap, location + new Vector3Int(-1, -1, 0)) ? 32 : 0;
+            mask += TileValue(tileMap, location + new Vector3Int(-1, 0, 0)) ? 64 : 0;
+            mask += TileValue(tileMap, location + new Vector3Int(-1, 1, 0)) ? 128 : 0;
+
+            byte original = (byte)mask;
+            if ((original | 254) < 255) { mask = mask & 125; }
+            if ((original | 251) < 255) { mask = mask & 245; }
+            if ((original | 239) < 255) { mask = mask & 215; }
+            if ((original | 191) < 255) { mask = mask & 95; }
+
+            Sprite[][] m_Sprites = GetSpritesAsOne();
+
+            int index = GetIndex((byte)mask);
+            if (index >= 0 && index < m_Sprites.Length && TileValue(tileMap, location))
+            {
+                tileData.sprite = GetRandomSprite(m_Sprites[index]);
+                tileData.transform = GetTransform((byte)mask);
+                tileData.color = Color.white;
+                tileData.flags = TileFlags.LockTransform | TileFlags.LockColor;
+                tileData.colliderType = Tile.ColliderType.Sprite;
+            }
+        }
+
+        private Sprite GetRandomSprite(Sprite[] spriteList) {
+            if (spriteList.Length == 0) {
+                return null;
+            }
+
+            int spriteIndex = Random.Range(0, spriteList.Length - 1);
+            return spriteList[spriteIndex];
+        }
+
+        private Sprite[][] GetSpritesAsOne() {
+            Sprite[][] m_Sprites = new Sprite[15][];
+            m_Sprites[0] = Filled;
+            m_Sprites[1] = ThreeSides;
+            m_Sprites[2] = TwoSidesAndOneCorner;
+            m_Sprites[3] = TwoAdjacentSides;
+            m_Sprites[4] = TwoOppositeSides;
+            m_Sprites[5] = OneSideAndTwoCorners;
+            m_Sprites[6] = OneSideAndOneLowerCorner;
+            m_Sprites[7] = OneSideAndOneUpperCorner;
+            m_Sprites[8] = OneSide;
+            m_Sprites[9] = FourCorners;
+            m_Sprites[10] = ThreeCorners;
+            m_Sprites[11] = TwoAdjacentCorners;
+            m_Sprites[12] = TwoOppositeCorners;
+            m_Sprites[13] = OneCorner;
+            m_Sprites[14] = Empty;
+            return m_Sprites;
+        }
+
+        private bool TileValue(ITilemap tileMap, Vector3Int position)
+        {
+            TileBase tile = tileMap.GetTile(position);
+            return (tile != null && tile == this);
+        }
+
+        private int GetIndex(byte mask)
```

Comment:
This and GetTransform below is used by the original Terrain Tile itself. Perhaps we could consider making RandomTerrainTile inherit from TerrainTile and try to share as much common functions as possible?